controllers/host/platformnetworks.go

```diff
@@ -5,72 +5,953 @@ package host
 
 import (
 	"context"
+	"fmt"
 	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/starlingx/inventory/v1/addresspools"
+	"github.com/gophercloud/gophercloud/starlingx/inventory/v1/networkAddressPools"
+	"github.com/gophercloud/gophercloud/starlingx/inventory/v1/networks"
 	perrors "github.com/pkg/errors"
 	starlingxv1 "github.com/wind-river/cloud-platform-deployment-manager/api/v1"
 	utils "github.com/wind-river/cloud-platform-deployment-manager/common"
+	"github.com/wind-river/cloud-platform-deployment-manager/controllers/common"
+	cloudManager "github.com/wind-river/cloud-platform-deployment-manager/controllers/manager"
 	v1info "github.com/wind-river/cloud-platform-deployment-manager/platform"
 	"k8s.io/apimachinery/pkg/types"
 	kubeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
 )
 
-// TODO(sriram-gn): All platform network reconciliation workflows will be implemented here.
-// Currently we are just setting status of Reconciled / InSync of every platform network
-// and address pool instances to true just to acknowledge these objects are accepted.
-func (r *HostReconciler) ReconcilePlatformNetworks(client *gophercloud.ServiceClient, instance *starlingxv1.Host, profile *starlingxv1.HostProfileSpec, host *v1info.HostInfo) error {
-	if !utils.IsReconcilerEnabled(utils.HostPlatformNetwork) {
-		return nil
+// makeRangeArray converts an array of range structs to an array of arrays where
+// the inner array contains two elements.  The first element is the range start
+// address and the second element is the range end address.  This is to align
+// with the system API formatting which represents a pair as an array of two
+// elements.
+func makeRangeArray(ranges []starlingxv1.AllocationRange) [][]string {
```

Comment:
these 2 method should be moved to a common utils file.

Reply:
Tried to move `makeRangeArray` and `compareRangeArrays` but then realized it causes circular import issue between `package v1` and `package common`, ie. `v1` imports `common` and to get `makeRangeArray` to work we have to import `v1` in `common` :)